### PR TITLE
(#3216) Implement ncids footer

### LINKED ISSUE #3216: Story: Implement Footer from DS onto C.Gov
## As a CMS user, I would like to be able to edit the footer when content needs to be updated.

> **ESTIMATE** TBD

We need to implement the footer from the design system within Drupal for Cancer.gov's enterprise website (www.cancer.gov).

### Notes
- The floaty back to top dies here.
- Actually color palette still needs to be determined, as well as how much hacking will be needed for overriding USWDS colors. [This should be complete, see comps: https://app.mural.co/t/publicissapient7269/m/publicissapient7269/1652804277768/6500e76eb861e2d170cef472c81ddf30d3807a44?sender=meganlake7790]
- English confirmed below, Spanish still in translation as of 6/22.

### English Links
- Above Footer - Back to Top Icon
   - BACK TO TOP
- Main Link Area
   - About
      - [About this Website](https://www.cancer.gov/about-website)
      - [en Español](https://www.cancer.gov/espanol)
      - [Reuse & Copyright](https://www.cancer.gov/policies/copyright-reuse)
      - [Social Media](https://www.cancer.gov/social-media)
   - Resources
      - [Contact Us](https://www.cancer.gov/contact)
      - [Publications](https://www.cancer.gov/publications)
      - [Dictionary of Cancer Terms](https://www.cancer.gov/publications/dictionaries/cancer-terms)
      - [Find a Clinical Trial](https://www.cancer.gov/about-cancer/treatment/clinical-trials/search)
   - Policies
      - [Accessibility](https://www.cancer.gov/policies/accessibility)
      - [FOIA](https://www.cancer.gov/policies/foia)
      - [Privacy & Security](https://www.cancer.gov/policies/privacy-security)
      - [Disclaimers](https://www.cancer.gov/policies/disclaimer)
      - [Vulnerability Disclosure](https://www.hhs.gov/vulnerability-disclosure-policy/index.html)
- Subscribe box
   - Sign up for email updates
      - Enter your email address
      - Sign up [button]
   - Email subscribe box should pass through the user's email to https://public.govdelivery.com/accounts/USNIHNCI/subscriber/new without additional topics or category filters for C.gov
      - If a user has previously subscribed, user will sent through to subscription update page
      - New users will be sent to the email confirmation page
- Organization/DOC Name Area (I have no better name for this)
   - National Cancer Institute
   - At the National Institute of Health
- Contact Us Area
   - [Live Chat](https://livehelp.cancer.gov/)
      - This also is a popup 
   - 1-800-4-CANCER (Telephone link)
   - NCIinfo@nih.gov (Mailto link)
   - Site Feedback
      - [Qualtrics Survey Link](https://nci.az1.qualtrics.com/jfe/form/SV_aeLLobt6ZeGVn5I)
- Follow Us Area
   - Facebook Icon : https://www.facebook.com/cancer.gov
   - Twitter Icon: https://twitter.com/thenci
   - Instagram Icon: https://www.instagram.com/nationalcancerinstitute/
   - YouTube Icon: https://www.youtube.com/NCIgov
   - LinkedIn Icon: https://www.linkedin.com/company/nationalcancerinstitute/
- Organization area (I have no better name for this)
   - [U.S. Department of Health and Human Services](https://www.hhs.gov)
   - [National Institutes of Health](http://www.nih.gov/)
   - [National Cancer Institute](https://www.cancer.gov/)
   - [USA.gov](https://usa.gov/)  

### Spanish Links 
- Above Footer - Back to Top Icon
   - Volver arriba
- Main Link Area
   - About: Quiénes somos
      - [Acerca de este sitio web](https://www.cancer.gov/espanol/acerca-sitio)
      - [In English](http://[www.cancer,gov/espanol](http://www.cancer.gov))
      - [Derechos de autor](https://www.cancer.gov/espanol/politicas/derechos-de-autor-y-uso)
      - [Redes sociales](https://www.cancer.gov/espanol/redes-sociales)
   - Resources: Recursos
      - [Contacto](https://www.cancer.gov/espanol/contactenos)
      - [Publicaciones](https://www.cancer.gov/espanol/publicaciones)
      - [Diccionario de cáncer](https://www.cancer.gov/espanol/publicaciones/diccionarios/diccionario-cancer)
      - Note - There is no 'Find a Clinical Trial' in Spanish right now, but there might be plans to translate an intro page to CTS in the future. 
   - Policies: Políticas
      - [Accesibilidad](https://www.cancer.gov/espanol/politicas/accesibilidad)
      - [Ley de Libertad de la Información (FOIA)](https://www.cancer.gov/espanol/politicas/foia)
      - [Confidencialidad y seguridad](https://www.cancer.gov/espanol/politicas/confidencialidad-seguridad)
      - [Aviso legal](https://www.cancer.gov/espanol/politicas/exoneracion)
      ---- Note: There are plans to update this page, including the URL and H1. It is not a priority so it won’t be for a few months. Need to keep this in mind for MVP 1 launch depending on timeline. 
      - [Política de divulgación de vulnerabilidad](https://www.hhs.gov/es/vulnerability-disclosure-policy/index.html)
- Subscribe box
   - Sign up for email updates: Reciba actualizaciones por correo electrónico
      - Enter your email address: Ingrese su correo electrónico
      - Sign up [button]: Suscríbase
- Organization/DOC Name Area (I have no better name for this)
   - National Cancer Institute: Instituto Nacional del Cáncer
   - At the National Institute of Health: de los Institutos Nacionales de la Salud
- Contact Us: Comuníquese 
   - [Chat en vivo](https://livehelp-es.cancer.gov/)
      - This also is a popup 
   - 1-800-422-6237 (Telephone link)
   - NCIinfo@nih.gov (Mailto link)
   - Site Feedback: Comentarios y sugerencias
      - [Qualtrics Survey Link](https://nci.az1.qualtrics.com/jfe/form/SV_aeLLobt6ZeGVn5I)
- Follow Us Area
   - Facebook Icon: https://www.facebook.com/cancer.govEspanol
   - Twitter Icon: https://twitter.com/NCIespanol
   - YouTube Icon: https://www.youtube.com/playlist?list=PLYKy4VbxNln7Cy6K2hZvwHLIZAElio5ai
- Organization area (I have no better name for this)
   - [Departamento de Salud y Servicios Humanos (en inglés)](https://www.hhs.gov)
   - [Institutos Nacionales de la Salud](https://salud.nih.gov/)
   - [Instituto Nacional del Cáncer](http://www.cancer.gov/espanol)
   - [USA.gov en español](https://www.usa.gov/espanol/)  

### Acceptance Criteria

```gherkin
Scenario: User clicks on Live Chat and a popup appears
Given user is on a C.Gov page
When user clicks on “Live Chat” in the footer
  Then live chat opens in a pop-up window 
```
```gherkin
Scenario: User clicks on Site Feedback and a qualtrics survey opens
Given user is on a C.Gov page
When user clicks on "Site Feedback" in the footer
  Then a qualtrics survey opens in a new window
```

```gherkin
Scenario: Spanish - User clicks on Live Chat and a popup appears
Given user is on a C.Gov espanol page
When user clicks on "Chat en vivo” in the footer
  Then the live chat spanish opens in a pop-up window 
```

```gherkin
Scenario: User clicks on Back To Top and they are taken to the top of the page
Given user has scrolled down from the top of the page
When user clicks on "Back To Top" (located in the bottom right corner, even when over the footer)
  Then the user is taken to the top of the page
```

```gherkin
Scenario: User submits subscribe form with no email and gets error (validation)
Given user is viewing the footer on a C.gov page 
  And user has not entered anything in the email subscribe box
    When user clicks "Sign Up"
      Then system displays an "Invalid email address" error above the subscribe box
      And the box is outlined with an error color
      And there is a solid bar beside the area with the same error color
```

```gherkin
Scenario: New User submits subscribe form with correct email and is taken to subscribe
Given user is viewing the footer on a C.gov page 
  And user has entered a valid email address into the subscribe box 
  And user is not an existing email subscriber
  When user clicks "sign up"
    Then the user is taken to the new subscriber's page (https://public.govdelivery.com/accounts/USNIHNCI/subscriber/new)
† NOTE: This is technically functionality of GovDelivery. So if we can show that there is nothing we can do to mess that up, 
then there is really just 1 scenario.
```

```gherkin
Scenario: Existing User submits subscribe form with correct email and is taken to update page†
Given user is viewing the footer on a C.Gov page
  And user has entered a valid email address into the subscribe box 
  And user is not an existing email subscriber
  When user clicks "sign up"
    Then the user is taken to the subscription update page
† NOTE: This is technically functionality of GovDelivery. So if we can show that there is nothing we can do to mess that up, 
then there is really just 1 scenario.
```
```gherkin
Scenario: Spanish - User submits subscribe form with no email and gets error (validation) -- need to add spanish validation error message
Given user is viewing the footer on a C.gov page 
  And user has not entered anything in the email subscribe box
  When user clicks "Suscríbase"
    Then system displays an "El correo electrónico no es válido" error above the subscribe box
    And the box is outlined with an error color
    And there is a solid bar beside the area with the same error color
```

```gherkin
Scenario: Spanish - New User submits subscribe form with correct email and is taken to subscribe
Given user is viewing the footer on a C.gov espanol page 
  And user has entered a valid email address into the subscribe box 
  And user is not an existing email subscriber
  When user clicks "Suscríbase"
    Then the user is taken to the new subscriber's espanol page (https://public.govdelivery.com/accounts/USNIHNCIESP/subscriber/new)
† NOTE: This is technically functionality of GovDelivery. So if we can show that there is nothing we can do to mess that up, 
then there is really just 1 scenario.
```

```gherkin
Scenario: Spanish - Existing User submits subscribe form with correct email and is taken to update page
Given user is viewing the footer on a C.gov espanol page 
  And user has entered a valid email address into the subscribe box 
  And user is not an existing email subscriber
  When user clicks "Suscríbase"
    Then the user is taken to the subscription update espanol page
† NOTE: This is technically functionality of GovDelivery. So if we can show that there is nothing we can do to mess that up, 
then there is really just 1 scenario.
```

```gherkin
Scenario: User is on mobile breakpoint and main area links collapse to headings
Given user is at the bottom of a cgov page on 767 and smaller (mobile)
  Then each of the links in main areas have collapsed into accordions
  And the heading of the accordion is the heading for that set of links
```

```gherkin
Scenario: User is on mobile breakpoint and can expand one main area heading at a time
Given user is at the bottom of a cgov page on 767 and smaller (mobile)
  When user clicks on a list header
    Then the accordion opens to reveal links
  When user clicks on a link in the list
    Then that page loads
  When user clicks on a list header again (of a list that is open)
    Then that accordion collapses
```

### Acceptance Criteria for Analytics
``` gherkin
Scenario: Footer link click
Given a user is on any page of cancer.gov
When the user clicks on a link in the footer section
Then there should be an NCIDataLayer Other event 
  And the event has the name “Footer:LinkClick”
  And the event has the the linkname “Footer:LinkClick”
  And this event should have the following data:
      | key      | value                                                                                                        |
      | linkText | Text of the link clicked.                                                                                    |
      |          | For social media icons, pass the aria-label value (i.e. Facebook, Twitter, etc.)                             |
      | location | Footer                                                                                                       |
      | section  | The value of the area name. Will reflect the heading above the text (i.e. About, Resources, Follow Us, etc.) |
```

``` gherkin
Scenario: Back To Top link click
Given a user is on any page of cancer.gov
When the user clicks on the Back To Top link
Then there should be an NCIDataLayer Other event 
  And the event has the name "RightEdge:LinkClick"
  And the event has the the linkname "RightEdge:LinkClick"
  And this event should have the following data:
    | key      | value        |
    | linkText | Back To Top  |
    | location | Right Edge   |
    | section  | Back To Top  |
```

``` gherkin
Scenario: Mobile Section Expand Click
Given a user is on any page of cancer.gov
Then there should be an NCIDataLayer Other event 
  And the event has the name “Footer:SectionExpand”
  And the event has the the linkname “Footer:SectionExpand”
  And this event should have the following data:
      | key      | value                                                                                                        |
      | linkText | Expand                                                                                                       |
      | location | Footer                                                                                                       |
      | section  | The value of the area name. Will reflect the heading above the text (i.e. About, Resources, Follow Us, etc.) |
```

``` gherkin
Scenario: Email Sign-Up Form Click
Given a user is on any page of cancer.gov
Then there should be an NCIDataLayer Other event 
  And the event has the name “Form:Start”
  And the event has the the linkname “Form:Start”
  And this event should have the following data:
      | key      | value       |
      | formType | EmailSignUp |
      | status   | Start       |
      | location | Footer      |
```

``` gherkin
Scenario: Email Sign-Up Form Submit
Given a user is on any page of cancer.gov
Then there should be an NCIDataLayer Other event 
  And the event has the name “Form:Complete”
  And the event has the the linkname “Form:Complete”
  And this event should have the following data:
      | key      | value       |
      | formType | EmailSignUp |
      | status   | Complete    |
      | location | Footer      |
```

``` gherkin
Scenario: Email Sign-Up Form Error
Given a user is on any page of cancer.gov
Then there should be an NCIDataLayer Other event 
  And the event has the name “Form:Error”
  And the event has the the linkname “Form:Error”
  And this event should have the following data:
      | key      | value       |
      | formType | EmailSignUp |
      | status   | Error       |
      | location | Footer      |
```
---

### Resources:
* NCIDS Tickets:
   * https://github.com/NCIOCPL/ncids/issues/202 
   * https://github.com/NCIOCPL/ncids/issues/254 
   * https://github.com/NCIOCPL/ncids/issues/284

---

### Tasks:
- [ ] Update the YAML content to match the correct content. Work with Megan there is some middle ground between what is on the ODE and what is what we want.
- [ ] update ncids-common to follow the new pattern once you make the lib/component folder.
- [ ] Make the component WITH TESTS
- [ ] Images maybe?

### PR DESCRIPTION
**Pull request by olitharp-nci**
Closes #3216 

ODE, this branches off inner pages, so only inner pages have the new footer for now
https://ncigovcdode480.prod.acquia-sites.com/about-cancer/coping/feelings

### CODE REVIEW

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/usa-footer/_usa-footer.scss [hunk 1]
```diff
@@ -0,0 +1,6 @@
+@import '@nciocpl/ncids-css/scss/components/usa-footer/nci-big';
+
+//TODO remove me don't let this pr get pushed with this in
```
Comment (olitharp-nci) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/usa-footer/_usa-footer.scss:3]: Calling this out. To be fixed with https://github.com/NCIOCPL/ncids/pull/500

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/usa-footer/usa-footer.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+	return targetTextContent.trim() || errorMsg;
+};
+
+/**
+ * Returns the section heading from the clicked link.
+ * @param event Click link event.
+ */
+const getSectionText = (event: Event): string => {
+	const target = event.target as HTMLAnchorElement;
```
Comment (olitharp-nci) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/usa-footer/usa-footer.ts:66]: need to update this to event.currentTarget

**Comment by bryanpizzillo** [comment]
Code looks good to me.

**Comment by mworrest** [comment]
@olitharp-nci are you able to make the caret icons turn sideways when the footer accordion is closed (then turn down when it is opened)? https://publicis.invisionapp.com/overview/Header---Footer-Wireframes-cl17806xjx3ho01x42xix33gc/screens

**Comment by olitharp-nci** [comment]
@mworrest 
I'm not sure when this bug was introduced, but it definitely should be rotating. I'll spin up a DS ticket and try to get it pushed through. I'll ping you on Slack when it's ready for review.

**Comment by meglake** [comment]
Noting a few items from testing:
* Email sign up doesn't work (opens a 'about:blank' page)
* Spanish footer has a few items that still need to be translated:
** 'Contact Us' to 'Comuníquese'?
** 'Follow us' to 'Síganos en'

Noting that 'Comentarios y sugerencias' link (Site Feedback) wraps to second line because of how long the text is. We'll use end of sprint demo to get approval from clients to leave as is or find a fix.

**Comment by bennettcc** [comment]
Adding @welshja's feedback from testing: @olitharp-nci, on the email sign-up box, the "start" event fires every time someone clicks into that box, so you can just pile up analytics events. Can we update that to only be called on the first click into that box after the page is loaded? This is not stated in the requirements above, but is a change we'd want to make to avoid pilling up analytics calls.

The other thing I noticed with analytics is that it is not firing at all on the Spanish page (https://ncigovcdode480.prod.acquia-sites.com/espanol/cancer/sobrellevar/sentimientos). Not sure if this just isn't setup, but let me know.

Everything else is working as expected.

**Comment by bennettcc** [comment]
Per Slack conversation with @welshja - the issues documented on 7/20 (stacking click events from the sign up box and spanish analytics not working) have been resolved. Testing passes on the ODE

cc @olitharp-nci

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Why has the package-lock been modified? Please remove.

**Review by bryanpizzillo — APPROVED** [review]

**Pull request merged**